command/ssh.go

```diff
@@ -0,0 +1,180 @@
+	}
+
+	input := strings.Split(args[0], "@")
+	if len(input) != 2 {
+		c.Ui.Error(fmt.Sprintf("Invalid parameter: %s", args[0]))
+		return 2
+	}
+
+	username := input[0]
+
+	ip, err := net.ResolveIPAddr("ip", input[1])
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error resolving IP Address: %s", err))
+		return 2
+	}
+
+	if role == "" {
+		role, err = setDefaultRole(client, ip.String())
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error setting default role: %s", err.Error()))
+			return 1
+		}
+		c.Ui.Output(fmt.Sprintf("Vault SSH: Role:'%s'\n", role))
+	}
+
+	data := map[string]interface{}{
+		"username": username,
+		"ip":       ip.String(),
+	}
+
+	keySecret, err := client.SSH().KeyCreate(role, data)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error getting key for SSH session:%s", err))
+		return 2
+	}
+
+	if keySecret.Data["key_type"].(string) == ssh.KeyTypeDynamic {
+		sshDynamicKey := string(keySecret.Data["key"].(string))
+		if len(sshDynamicKey) == 0 {
+			c.Ui.Error(fmt.Sprintf("Invalid key"))
+			return 2
+		}
+		sshDynamicKeyFileName = fmt.Sprintf("vault_temp_file_%s_%s", username, ip.String())
+		err = ioutil.WriteFile(sshDynamicKeyFileName, []byte(sshDynamicKey), 0600)
+		sshCmdArgs = append(sshCmdArgs, []string{"-i", sshDynamicKeyFileName}...)
+
+	} else if keySecret.Data["key_type"].(string) == ssh.KeyTypeOTP {
+		fmt.Printf("OTP for the session is %s\n", string(keySecret.Data["key"].(string)))
+	} else {
+		// Intentionally not mentioning the exact error
+		c.Ui.Error("Error creating key")
+	}
+	sshCmdArgs = append(sshCmdArgs, []string{"-p", port}...)
+	sshCmdArgs = append(sshCmdArgs, args...)
+
+	sshCmd := exec.Command("ssh", sshCmdArgs...)
+	sshCmd.Stdin = os.Stdin
+	sshCmd.Stdout = os.Stdout
+
+	err = sshCmd.Run()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error while running ssh command:%s", err))
+	}
+
+	if keySecret.Data["key_type"].(string) == ssh.KeyTypeDynamic {
+		err = os.Remove(sshDynamicKeyFileName)
+		if err != nil {
+			// Intentionally not mentioning the exact error
+			c.Ui.Error("Error cleaning up")
+		}
+	}
+
+	err = client.SSH().KeyRevoke(keySecret.LeaseID)
+	if err != nil {
+		// Intentionally not mentioning the exact error
+		c.Ui.Error("Error cleaning up")
+	}
+
+	return 0
+}
+
+func setDefaultRole(client *api.Client, ip string) (string, error) {
+	data := map[string]interface{}{
+		"ip": ip,
+	}
+	secret, err := client.Logical().Write("ssh/lookup", data)
+	if err != nil {
+		return "", fmt.Errorf("Error finding roles for IP '%s':%s", ip, err)
+
+	}
+	if secret == nil {
+		return "", fmt.Errorf("Error finding roles for IP '%s':%s", ip, err)
+	}
+
+	if secret.Data["roles"] == nil {
+		return "", fmt.Errorf("IP '%s' not registered under any role", ip)
+	}
+
+	if len(secret.Data["roles"].([]interface{})) == 1 {
+		return secret.Data["roles"].([]interface{})[0].(string), nil
+	} else {
+		return "", fmt.Errorf("Multiple roles for IP '%s'. Select one of '%s' using '-role' option", ip, secret.Data["roles"])
```

Comment:
We should provide a nicer output, and show this as a list of possible roles and advise that not all may be permitted based on ACLs